task-09/main.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       Example application for demonstrating the RIOT network stack
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "shell.h"
+#include "msg.h"
+#include "thread.h"
+
+#include "net/gnrc.h"
+#include "net/gnrc/ipv6/netif.h"
+#include "net/gnrc/ipv6.h"
+
+#define MAIN_QUEUE_SIZE     (8)
+static msg_t _main_msg_queue[MAIN_QUEUE_SIZE];
+
+extern int udp_cmd(int argc, char **argv);
+
+static const shell_command_t shell_commands[] = {
+    { "udp", "send data over UDP and listen on UDP ports", udp_cmd },
+    { NULL, NULL, NULL }
+};
+
+#ifdef MODULE_GNRC_SIXLOWPAN
+static char _stack[THREAD_STACKSIZE_MAIN];
+
+static void *_ipv6_fwd_eventloop(void *arg)
+{
+    (void)arg;
+
+    msg_t msg, msg_q[8];
+    gnrc_netreg_entry_t me_reg;
+
+    msg_init_queue(msg_q, 8);
+
+    me_reg.demux_ctx = GNRC_NETREG_DEMUX_CTX_ALL;
+    me_reg.pid = thread_getpid();
+
+    gnrc_netreg_register(GNRC_NETTYPE_SIXLOWPAN, &me_reg);
+
+    while(1) {
+        msg_receive(&msg);
+        gnrc_pktsnip_t *pkt = msg.content.ptr;
+        if(msg.type == GNRC_NETAPI_MSG_TYPE_SND) {
+            ipv6_hdr_t *ipv6_hdr = pkt->next->data;
```

Comment:
using `LL_SEARCH` to search for the ipv6 packet may be more appropriate, in case there is something in between?